src/diagnostic.c

```diff
@@ -198,6 +198,7 @@ static const char* const diagnostic_messages[PM_DIAGNOSTIC_ID_LEN] = {
     [PM_ERR_OPERATOR_WRITE_BLOCK]               = "Unexpected operator after a call with a block",
     [PM_ERR_PARAMETER_ASSOC_SPLAT_MULTI]        = "Unexpected multiple `**` splat parameters",
     [PM_ERR_PARAMETER_BLOCK_MULTI]              = "Multiple block parameters; only one block is allowed",
+    [PM_ERR_PARAMETER_CIRCULAR]                 = "Invalid circular reference in a default parameter",
```

Comment:
I am not sure here "Invalid" sounds like good wording.

Reply:
```suggestion
    [PM_ERR_PARAMETER_CIRCULAR]                 = "Parameter default value references itself",
```